### ISSUE TITLE
loss issues encountered in fine-tuning the model

### ISSUE THREAD

**Issue by wangtao201919** [bshall/knn-vc#26]
Hello author, this project is great. I am trying to add some Chinese speeches for fine-tuning, but my ’validation/mel_spec_error‘ has almost stopped decreasing at 15k, and ’training/gen_loss_total' has also increased. I would like to ask if this loss change is normal. Thank you so much.
![11](https://github.com/bshall/knn-vc/assets/46395380/2a23803d-dd5b-46ee-8013-80b6dfb6d546)

**Comment by RF5** [comment]
Hi @wangtao201919 

The values of the training loss curves do look a little strange, the generator loss probably should be increasing so much. But it is a little hard to say since the training loss dynamics of GANs can be a bit tricky to interpret. The validation mel spec error also ideally should be decreasing, but again it is a bit hard to tell. My advice is to listen to the samples generated every 20k steps or so and assess whether it sounds better or not. 

To give some comparison, here is what the training curves look like for our prematched vocoder trained on librispeech for the first 1M steps:

![image](https://github.com/bshall/knn-vc/assets/23717819/8d08eac9-d1ff-40f7-9971-5f22f9a7c6b7)

Aside from the generator loss, our plots and yours look fairly similar. 

I hope that helps a bit!

**Comment by wangtao201919** [comment]
Thank you very much for your response and suggestions. I have tested the fine-tuning model, and the results are very significant. However, I have found that the generalization in cross-lingual conversion is not stable enough, just as you mentioned in your paper, "how far away can the reference utterances be from the training distribution?" I suspect that it might be because the distance between the features of the target language and the reference language cannot be measured as accurately as within the same language.

**Comment by RF5** [comment]
Ahh I see. Yeah for languages that are very very different from English, one might need to fine-tune the WavLM encoder as well on that language to allow it to better represent it in the feature space. Without this, you are probably right that the distance comparisons between features of different languages is not as ideal as within the same language.

**Comment by egorsmkv** [comment]
@RF5 is there any tutorial to fine-tune the WavLM encoder?

**Comment by RF5** [comment]
Hi @egorsmkv , unfortunately not that we know of. There might be some useful resources for that in the [original microsoft repo](https://github.com/microsoft/unilm/tree/master/wavlm), but I don't think they ever open sourced their training code.

**Comment by EmreOzkose** [comment]
Hi @wangtao201919 , did you fine-tune wavLM for Chinese ? If not, did you obtain good results with just fine-tuning vocoder?
